plugin/trino-hive/src/test/java/io/trino/plugin/hive/HadoopQueryRunner.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.log.Logger;
+import io.airlift.log.Logging;
+import io.trino.Session;
+import io.trino.plugin.tpch.TpchPlugin;
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.SelectedRole;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.tpch.TpchTable;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.airlift.testing.Closeables.closeAllSuppress;
+import static io.trino.plugin.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+import static io.trino.spi.security.SelectedRole.Type.ROLE;
+import static io.trino.testing.QueryAssertions.copyTpchTables;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+
+public final class HadoopQueryRunner
+{
+    private static final SelectedRole ADMIN_ROLE = new SelectedRole(ROLE, Optional.of("admin"));
+
+    private HadoopQueryRunner() {}
+
+    static {
+        System.setProperty("HADOOP_USER_NAME", "hive");
+    }
+
+    public static DistributedQueryRunner createHadoopQueryRunner(
+            TestingHadoopServer server,
+            Map<String, String> extraProperties,
+            Map<String, String> connectorProperties,
+            Iterable<TpchTable<?>> tables)
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(createSession())
+                .setExtraProperties(extraProperties)
+                .build();
+        try {
+            queryRunner.installPlugin(new TpchPlugin());
+            queryRunner.installPlugin(new TestingHadoopPlugin());
```

Comment:
Why not use pre-exising io.trino.plugin.hive.HiveQueryRunner ?
```
 return HiveQueryRunner.builder()
    .with......
    .build()
```

Reply:
I avoided reusing `HiveQueryRunner` intentionally because the class uses file based HMS and preparing Thrift HMS object reduces readability.